### ISSUE TITLE
iOS not building on develop (45fc937419069a94dab3d4dd11deab4f33b080a0)

### ISSUE THREAD

**Issue by tushar-dadlani** [tripleblindmarket/private-kit#65]
I am getting this error on the iOS build for develop https://github.com/tripleblindmarket/private-kit/commit/45fc937419069a94dab3d4dd11deab4f33b080a0

It gets stuck on a white screen and the app fails to load. I tested release v0.5 and it worked.

**Device iPhone 8 - iOS 13.3.1**
```
2020-03-16 01:01:35.631 [info][tid:main][RNGestureHandlerManager.m:136] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x1013049b0; reactTag: 1; frame = (0 0; 375 667); gestureRecognizers = <NSArray: 0x2800f1170>; layer = <CALayer: 0x280eb8ec0>>
2020-03-16 01:01:35.660749-0700 PrivateKit[438:28526] +backgroundSessionConfiguration: is deprecated. Please use +backgroundSessionConfigurationWithIdentifier: instead
2020-03-16 01:01:35:660 PrivateKit[438:28526] Backing up original handler.
2020-03-16 01:01:35:660 PrivateKit[438:28526] Setting new handler.
2020-03-16 01:01:35.693860-0700 PrivateKit[438:28526] Current db version: 3
2020-03-16 01:01:35:694 PrivateKit[438:28519] BgGeo #configure: Config: distanceFilter=50 stationaryRadius=50 desiredAccuracy=0 activityType=OtherNavigation activitiesInterval=10000 isDebugging=0 stopOnTerminate=0 url= syncThreshold=100 maxLocations=10000 httpHeaders={
} pauseLocationUpdates=0 saveBatteryOnBackground=0 locationProvider=1 postTemplate={
    foo = bar;
    lat = "@latitude";
    lon = "@longitude";
}
2020-03-16 01:01:35.695266-0700 PrivateKit[438:28523] BgGeo #configure: Config: distanceFilter=50 stationaryRadius=50 desiredAccuracy=0 activityType=OtherNavigation activitiesInterval=10000 isDebugging=0 stopOnTerminate=0 url= syncThreshold=100 maxLocations=10000 httpHeaders={
} pauseLocationUpdates=0 saveBatteryOnBackground=0 locationProvider=1 postTemplate={
    foo = bar;
    lat = "@latitude";
    lon = "@longitude";
}
2020-03-16 01:01:35.696687-0700 PrivateKit[438:28519] Configuration persisted succesfuly
2020-03-16 01:01:35.700 [error][tid:com.facebook.react.JavaScript] TypeError: s.registerHeadlessTask is not a function. (In 's.registerHeadlessTask(o,n)', 's.registerHeadlessTask' is undefined)

This error is located at:
    in n
    in Unknown
    in Unknown
    in Unknown
    in Unknown
    in RCTView
    in RCTView
    in RCTView
    in RCTView
    in u
    in PanGestureHandler
    in RCTView
    in u
    in RCTView
    in n
    in Unknown
    in RCTView
    in j
    in RCTView
    in C
    in o
    in n
    in RNCSafeAreaView
    in Unknown
    in Unknown
    in RCTView
    in n
    in Unknown
    in Unknown
    in ForwardRef
    in Unknown
    in ForwardRef
    in n
    in n
    in RCTView
    in RCTView
    in h
```

**Comment by vitorpamplona** [comment]
Same issue here. Trying to figure out. Probably something related to the headlessTask

https://github.com/mauron85/react-native-background-geolocation#android-headless-task-experimental

**Comment by vitorpamplona** [comment]
I think I resolved this on Branch #57 that also includes @Andy's warning fixes.

**Comment by vitorpamplona** [comment]
Solved with #67
